### DISCUSSION TITLE
Input delay in Lazer for MacOS

### DISCUSSION THREAD

**Discussion by kashsuks** [ppy/osu#30351] [Ideas] [answered]
Hi guys,

I'm an osu! player who used to play stable on my windows machine. After I switched to mac for unix development I decided to try playing stable. Firstly, I couldn't get stable to work properly without constant frame drops due to having to use wineskin. I then tried to Lazer, which wasn't too bad but then I realized that my input delay was hovering around 8ms which is a lot more compared to the 0.5 I got on stable. Any idea how to help fix this, or if there already is a fix?

**Comment by peppy** [comment]
What is this 8ms you are seeing? macOS uses hardware level frame sync so input handling should be pretty good.

**Comment by kashsuks** [comment]
I can provide you with some gameplay footage.

Below are the specs of my machine:

M3 Macbook Pro
18GB Unified RAM
512GB SSD


https://github.com/user-attachments/assets/0d445853-24e9-45a9-af99-3058f3c1065a

**Reply by kashsuks** [reply]
In the bottom right corner of the footage, there is the fps counter and what im assuming is the input delay. This same stat in stable was under 1 ms

**Reply by Joehuu** [reply, marked as answer]
It seems like you are on singlethreaded for some reason. Does changing to multithreaded (default) help?

**Reply by kashsuks** [reply]
Ah yes, using multithreaded rendering does help. It goes from 9ms -> 2.2ms

**Reply by peppy** [reply]
Numbers don't mean much here. the 9 ms will be more correctly synced, in theory.
